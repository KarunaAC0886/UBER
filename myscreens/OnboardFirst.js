import { View, Text, SafeAreaView, TouchableOpacity, TextInput, StatusBar } from 'react-native'
import React, { useState } from 'react'
import { useNavigation } from '@react-navigation/native';
import styles from './styles'
import OnboardBackNavigator from './OnboardBackNavigator';


const OnboardFirst = () => {

    const [firstname, setFirstname] = useState('')
    const [lastname, setLastname] = useState('')
    const [firstNameError, setFirstNameError] = useState('');
    const [lastNameError, setLastNameError] = useState('');

    const firstnameValidation = (value)=>{
        setFirstname(value);
        const firstNameRegex = /^[A-Za-z]+(?:['`-][A-Za-z]+)*$/;
        if (firstname == "") {
            setFirstNameError("please fill firstname");
        }else if(firstNameRegex.test(firstname)){
            setFirstNameError("")
        }else{
            setFirstNameError("it must start with string")
        }
    }

    
    const lastnameValidation = (value)=>{
        setLastname(value);
        const firstNameRegex = /^[A-Za-z]+(?:['`-][A-Za-z]+)*$/;
        if (lastname == "") {
            setLastNameError("please fill lastname");
        }else if(firstNameRegex.test(lastname)){
            setLastNameError("")
        }else{
            setLastNameError("it must start with string")
        }
    }
    const NameValide = () => {
        
        if (firstname == "" && lastname == "") {
            setFirstNameError("please fill firstname");
            setLastNameError("please fill lastname")
        } else if (lastname == "") {
            setLastNameError("please fill lastname");
        } else {
            setFirstname("");
            setLastname("")
            navigation.navigate('onboardprofile')

        }
    }

    const navigation = useNavigation();
    return (
        <SafeAreaView style={styles.onb_frst_mn_cntr}>

            <View style={styles.onb_nms_hdr}>
                <OnboardBackNavigator navpath="onboardphonenumber" />
            </View>

            <View style={styles.onb_nms_cntr}>
                <Text style={styles.onb_nms_nmtxtfld}>What's Your Name?</Text>

                <View style={styles.onb_nm_nms_whlocntr}>
                    <View>
                        <TextInput style={styles.onb_nm_nms_frst} value={firstname} onChangeText={firstnameValidation} placeholder='First name'/>
                        <Text style={{color:'red',fontSize:16}}>{firstNameError}</Text>
                    </View>
                    <View>
                        <TextInput style={styles.onb_nm_nms_lst} value={lastname} onChangeText={lastnameValidation} placeholder='Last name'/>
                        <Text style={{color:'red',fontSize:16}}>{lastNameError}</Text>
                    </View>
                </View>

                <TouchableOpacity style={styles.onb_nm_nxtbtncont} onPress={NameValide}>
                    <Text style={styles.onb_nm_nxtlbl} >Next</Text>
                </TouchableOpacity>
            </View>
            <StatusBar />
        </SafeAreaView>
    )
}
export default OnboardFirst