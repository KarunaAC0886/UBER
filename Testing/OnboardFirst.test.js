

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import OnboardFirst from '../myscreens/OnboardFirst';

jest.mock('@expo/vector-icons', () => ({
    AntDesign: 'AntDesign', 
}));


jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

jest.mock('react-native-responsive-screen', () => ({
  widthPercentageToDP: jest.fn(),
  heightPercentageToDP: jest.fn(),
}));

describe('First component', () => {
  it('renders correctly', () => {
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    const tree = renderer.create(<OnboardFirst/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

