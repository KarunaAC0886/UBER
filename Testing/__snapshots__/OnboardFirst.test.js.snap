// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First component renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#1A1A1A",
    }
  }
>
  <View
    style={
      {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#1A1A1A",
          "height": "100%",
          "justifyContent": "center",
          "paddingLeft": "6%",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <AntDesign
          name="arrowleft"
          style={
            {
              "color": "#ffffff",
              "fontSize": 23,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "height": undefined,
        "paddingHorizontal": undefined,
        "width": undefined,
      }
    }
  >
    <Text
      style={
        {
          "color": "#EDF6FF",
          "fontSize": 30,
          "height": 33,
        }
      }
    >
      What's Your Name?
    </Text>
    <View
      style={
        {
          "flex": 0,
          "flexDirection": "row",
          "marginTop": undefined,
        }
      }
    >
      <View>
        <TextInput
          onChangeText={[Function]}
          placeholder="First name"
          style={
            {
              "borderBottomWidth": 1,
              "borderColor": "#EDF6FF",
              "color": "#EDF6FF",
              "fontSize": 20,
              "width": undefined,
            }
          }
          value=""
        />
        <Text
          style={
            {
              "color": "red",
              "fontSize": 16,
            }
          }
        />
      </View>
      <View>
        <TextInput
          onChangeText={[Function]}
          placeholder="Last name"
          style={
            {
              "borderBottomWidth": 1,
              "borderColor": "#EDF6FF",
              "color": "#EDF6FF",
              "fontSize": 20,
              "marginLeft": "5%",
              "width": undefined,
            }
          }
          value=""
        />
        <Text
          style={
            {
              "color": "red",
              "fontSize": 16,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderColor": "#EDF6FF",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 58,
          "justifyContent": "center",
          "marginTop": undefined,
          "opacity": 1,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 21,
            "left": 150,
          }
        }
      >
        Next
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
